packages/edge/src/edge.ts

```diff
@@ -0,0 +1,219 @@
+import { encode } from "@msgpack/msgpack";
+
+import {
+  Context,
+  ILogLevel,
+  ILogtailLog,
+  ILogtailOptions,
+  LogLevel,
+} from "@logtail/types";
+import { Base } from "@logtail/core";
+
+import { ExecutionContext } from "@cloudflare/workers-types";
+
+import { getStackContext } from "./context";
+import { EdgeWithExecutionContext } from "./edgeWithExecutionContext";
+
+// Types
+type Message = string | Error;
+
+export class Edge extends Base {
+  private _warnedAboutMissingCtx: Boolean = false;
+
+  public constructor(sourceToken: string, options?: Partial<ILogtailOptions>) {
+    super(sourceToken, options);
+
+    // Sync function
+    const sync = async (logs: ILogtailLog[]): Promise<ILogtailLog[]> => {
+      const res = await fetch(this._options.endpoint, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/msgpack",
+          Authorization: `Bearer ${this._sourceToken}`,
+          "User-Agent": "logtail-js(edge)",
+        },
+        body: this.encodeAsMsgpack(logs),
+      });
+
+      if (res.ok) {
+        return logs;
+      }
+
+      throw new Error(res.statusText);
+    };
+
+    // Set the throttled sync function
+    this.setSync(sync);
+  }
+
+  public withExecutionContext(ctx: ExecutionContext): EdgeWithExecutionContext {
+    return new EdgeWithExecutionContext(this, ctx);
+  }
+
+  /**
+   * @param message (string | Error) - Log message
+   * @param level (ILogLevel) - Level to log at (debug|info|warn|error)
+   * @param context (Context | Error | any) - Log context for passing structured data
+   * @param ctx (ExecutionContext) - Execution context of particular worker request
+   * @returns Promise<ILogtailLog> after syncing
+   */
+  public async log<TContext extends Context>(
+    message: Message,
+    level?: ILogLevel,
+    context: any = {} as TContext,
+    ctx?: ExecutionContext,
+  ): Promise<ILogtailLog & TContext> {
+    // Wrap context in an object, if it's not already
+    if (typeof context !== "object") {
+      const wrappedContext: unknown = { extra: context };
+      context = wrappedContext as TContext;
+    }
+    if (context instanceof Error) {
+      const wrappedContext: unknown = { error: context };
+      context = wrappedContext as TContext;
+    }
+
+    // Process/sync the log, per `Base` logic
+    context = { ...getStackContext(this), ...context };
+    const log = super.log(message, level, context);
+
+    if (ctx) {
+      ctx.waitUntil(log);
+    } else if (!this._warnedAboutMissingCtx) {
+      this._warnedAboutMissingCtx = true;
+      console.warn(
+        "ExecutionContext hasn't been set via `withExecutionContext` method. Logs may not reach Better Stack unless you manually call `ctx.waitUntil(log)`.",
+      );
```

Comment:
Is there any reason why people might not want to use the `withExecutionContext`? E.g. maybe there could be workers that doesn't make the context easily available?

I would consider throwing an exception if there is never a good reason to not use `withExecutionContext`.
But it's still a little weird for logging library to bring down people's worker code.

How about send the warning to both console and Better Stack? Plus flush after the warning to make sure people at least see the warning in Live tail. I imagine seeing the warning instead of actual logs in Live tail could work well.

Also could you make sure that the warning gives people concrete instructions how to fix the issue? 🙏
E.g. now I feel like the warning tells me that the easiest solution is to add `ctx.waitUntil(log)` to my worker code. But the recommended way is to use `withExecutionContext()`.
Could you change the warning to make it clearer how to fix the issue? Link to docs page might work well. Thanks!